crates/bevy_gizmos/src/gizmos.rs

```diff
@@ -14,43 +14,93 @@ use bevy_transform::TransformPoint;
 type PositionItem = [f32; 3];
 type ColorItem = [f32; 4];
 
+/// Storage of gizmo primitives.
 #[derive(Resource, Default)]
-pub(crate) struct GizmoStorage {
-    pub list_positions: Vec<PositionItem>,
-    pub list_colors: Vec<ColorItem>,
-    pub strip_positions: Vec<PositionItem>,
-    pub strip_colors: Vec<ColorItem>,
+pub struct GizmoStorage<Clear> {
+    pub(crate) list_positions: Vec<PositionItem>,
+    pub(crate) list_colors: Vec<ColorItem>,
+    pub(crate) strip_positions: Vec<PositionItem>,
+    pub(crate) strip_colors: Vec<ColorItem>,
+    _phantom: PhantomData<Clear>,
 }
 
+impl<Clear> GizmoStorage<Clear>
+where
+    Clear: 'static + Send + Sync + Default,
+{
+    /// Combine the other gizmo storage with this one.
+    pub fn append_storage<OtherClear>(&mut self, other: &GizmoStorage<OtherClear>) {
+        self.list_positions.extend(other.list_positions.iter());
+        self.list_colors.extend(other.list_colors.iter());
+        self.strip_positions.extend(other.strip_positions.iter());
+        self.strip_colors.extend(other.strip_colors.iter());
+    }
+
+    pub(crate) fn swap<OtherClear>(&mut self, other: &mut GizmoStorage<OtherClear>) {
+        mem::swap(&mut self.list_positions, &mut other.list_positions);
+        mem::swap(&mut self.list_colors, &mut other.list_colors);
+        mem::swap(&mut self.strip_positions, &mut other.strip_positions);
+        mem::swap(&mut self.strip_colors, &mut other.strip_colors);
+    }
+
+    /// Clear this gizmo storage of any requested gizmos.
+    pub fn clear(&mut self) {
+        self.list_positions.clear();
+        self.list_colors.clear();
+        self.strip_positions.clear();
+        self.strip_colors.clear();
+    }
+}
+
+/// Clear each [`Update`].
+///
+/// Default context for [`Gizmos`].
+#[derive(Default)]
+pub struct Update;
+
+/// Swap buffer for gizmos.
+#[derive(Default)]
+pub struct Swap;
+
 /// A [`SystemParam`] for drawing gizmos.
 ///
 /// They are drawn in immediate mode, which means they will be rendered only for
 /// the frames in which they are spawned.
```

Comment:
Is this better?

"They are drawn in immediate mode, which means they will be rendered only for
the frames, or ticks in [`FixedMain`](bevy_app::FixedMain)'s case, in which
they are spawned."

I'm trying to not get too in the weeds with it so that users don't get confused.